cmd/acr/purge.go

```diff
@@ -264,10 +279,22 @@ func getMatchingRepos(ctx context.Context, repoNames []string, repoRegex string)
 
 // getRepositoryAndTagRegex splits the strings that are in the form <repository>:<regex filter>
 func getRepositoryAndTagRegex(filter string) (string, string, error) {
-	repoAndRegex := strings.Split(filter, ":")
+	// This only selects colons that are not apart of a non-capture group
```

Comment:
I think I don't have the full context on how a non capture group can be used here, do you mind explaining it a little bit?

Reply:
Sure, A non-capture group in regex is `(?:)`. When you specify a filter for purge you need to format it as <repo-name>:<tag>. Previously the code split on ':', but since non-capture groups include a ':' in it's syntax, split would split incorrectly if a non-capture group was included. See https://github.com/Azure/acr-cli/issues/96